backend/GameManager.py

```diff
@@ -1,17 +1,41 @@
 # import pygame as pg
 from backend.Board import Board
-from backend.Cell import CellType
-from backend.Piece import Pieces, Turn
+from backend.Piece import Pieces
+from backend.Cell import Color
 
 
 class GameManager:
     def __init__(self, size, pieces):
         self.board = Board(size)
         self.pieces = Pieces(pieces)
-        self.turn = Turn.WHITE
+        self.turn = Color.WHITE
+        self.selected_piece = None
+        self.open_moves = self.board.get_valid_moves()
+        
+        # Mill variables
+        self.mills = [
+            [(0, 0), (0, 3), (0, 6)],  # Horizontal mills
+            [(1, 1), (1, 3), (1, 5)],
+            [(2, 2), (2, 3), (2, 4)],
+            [(3, 0), (3, 1), (3, 2)],
+            [(3, 4), (3, 5), (3, 6)],
+            [(4, 2), (4, 3), (4, 4)],
+            [(5, 1), (5, 3), (5, 5)],
+            [(6, 0), (6, 3), (6, 6)],
+
+            # Vertical mills
+            [(0, 0), (3, 0), (6, 0)],
+            [(1, 1), (3, 1), (5, 1)],
+            [(2, 2), (4, 2), (5, 2)],
+            [(0, 3), (1, 3), (2, 3)],
+            [(4, 3), (5, 3), (6, 3)],
+            [(2, 4), (4, 4), (5, 4)],
+            [(1, 5), (3, 5), (5, 5)],
+            [(0, 6), (3, 6), (6, 6)]
+        ]
+        self.waiting_for_removal = False
+        self.removable_pieces = []
 
```

Comment:
We should have a GamePhase variable in this class or as its own class to track the phase that we are in. Placement, Movement, MillFormed, Fly, GameOver.

[Example Enum](https://github.com/CS-449-Group-Project/NineMensMorris/blob/master/src/Morris_FX/Logic/GameManager.java#L26-L32)

Reply:
all phases are in consistent, ex : mill can be formed even in placement phase so need to check on that, 

changing conditions like this is okay but not sure how we can do this in our system but yes this will benefit a lot.